internal/cmd/skupper/site/kube/site_create.go

```diff
@@ -210,7 +210,7 @@ func (cmd *CmdSiteCreate) WaitUntilReady() error {
 			return err
 		}
 
-		if resource != nil && resource.Status.StatusMessage == "OK" {
+		if resource != nil && resource.IsActive() {
```

Comment:
@nluaces @ssorj at present, there are (up to) three conditions associated with site status. The site is considered successfully 'configured'  once the various resources have been created. If the site accepts incoming links (i.e. linkAccess is not 'none', which is the default) the site is considered successfully 'resolved' once the hosts/ports through which it can be linked to are discovered. The site is considered to be 'running' when there is at least one running router pod.

The site is currently considered 'ready' when it is configured, resolved if linkAccess is not none, and running.

The IsActive() function was mostly intended to be used in transitioning from old status to new. It just tests the configured condition. So using that method here is certainly in line with what was there before and not 'wrong'. However, as the function here is WaitUntilReady() I felt it worth starting a conversation on the meaning of 'ready' so we can eventually align.

Is the current site definition of 'ready' what is wanted here? In the case that it waits for resolution, or the images to be pulled, that can take some time. However, once ready then the site is much more likely to be in the usable state the the user would likely expect.

Reply:
@grs @ssorj some time ago, we discussed that after creating a site, it should be ready for the user to operate freely with it, that was the purpose to make the user wait. In principle, I think that the WaitUntilReady() should wait for the site to pass the isReady condition. I have tested this on my local setup: 

```
~/demo2 » time skupper site create noe --enable-link-access -n north                                  
Waiting for status...
Site "noe" is ready
skupper site create noe --enable-link-access -n north  0.10s user 0.08s system 0% cpu 25.067 total
-----------------------------------------------------------------------------------------------------------------------
~/demo2 » time skupper site create eon --enable-link-access -n south                                  
Waiting for status...
Site "eon" is ready
skupper site create eon --enable-link-access -n south  0.13s user 0.09s system 0% cpu 33.071 total
```
Some times it takes around 25~30 seconds, and other times when my minikube is not working properly, it could take more than 1 minute (which is something that the user may face and subsequently think that skupper is responsible for that). 

I also do think we need to check the existence of a site that is ready with the rest of the commands that rely on that assumption. 

Option A:
- Wait until the site is just configured (isActive)
- Rename `WaitUntilReady` to `WaitUntil`
- Output: "Site "my-site" is configured. Check the status to see when it is ready."
- Check that the site is ready in the rest of commands

Option B:
- Wait until the site is ready after its creation (timeout: 120s)
- Check that the site is ready in the rest of commands

Which option do you prefer? Do you have any other ideas?

Reply:
I like option B.    I also think that we could change the CLI status commands, right now we just show status OK,
Maybe we could show status as configured/resolved/ready.   Right now I always use the kubectl command to 
get status.

Reply:
I like option A.    Have the command wait until configured.   Then when user issues listener/connector commands we check that the site is Ready before allowing command.

I think we need to do the same with the listener/connector commands, I'm currently waiting for the status to show configured and return success.

Reply:
In pkg/apis/skupper/v1alpha1/types.go Should we change IsActive to IsConfigured for the Site?

Reply:
I think I prefer A as well

Reply:
fixed: a958b42f8caa260a955778454ed2b4404992512c